src/core/platform.h

```diff
@@ -81,7 +87,7 @@ struct IrqController {
         std::queue<IrqHandler> one_time_subscribers{};
         std::mutex m_lock{};
     };
-    std::array<IrqContext, magic_enum::enum_count<InterruptId>()> irq_contexts{};
+    std::unordered_map<InterruptId, IrqContext> irq_contexts{};
```

Comment:
The number of interrupts is still quite small so an array will be faster than unoredered_map

Reply:
Before this change the array was indexed with InterruptId, now it was changed to hash map, so unless there is a new behavior that relies on the hashmap I don't see why abandoning array is needed

Reply:
Can InterruptIdMax be increased to 0x41 so it works